src/ip_notifier.cpp

```diff
@@ -130,7 +130,11 @@ struct ip_change_notifier_impl final : ip_change_notifier
 			[](SCNetworkReachabilityRef /*target*/, SCNetworkReachabilityFlags /*flags*/, void *info)
 			{
 				auto obj = static_cast<ip_change_notifier_impl*>(info);
-				obj->m_ios.post([obj]() { if (obj->m_cb) obj->m_cb(error_code()); });
+				if (!obj->m_cb) return;
```

Comment:
I'm not an expert in dispatch_queue, but I'm sure it is another thread/pool.  I see your point with the multithread issue :) let me do that right now